velox/serializers/PrestoSerializer.cpp

```diff
@@ -1662,10 +1676,93 @@ class PrestoVectorSerializer : public VectorSerializer {
     out->seekp(offset + size);
   }
 
+  void flushCompressed(
+      int32_t numRows,
+      bool rle,
+      OutputStream* output,
+      PrestoOutputStreamListener* listener) {
+    const int32_t offset = output->tellp();
+    char codec = kCompressedBitMask;
+    if (listener) {
+      codec |= kCheckSumBitMask;
+    }
+
+    // Pause CRC computation
+    if (listener) {
+      listener->pause();
+    }
+
+    writeInt32(output, numRows);
+    output->write(&codec, 1);
+
+    IOBufOutputStream out(
+        *(streamArena_->pool()), nullptr, streamArena_->size());
+    writeInt32(&out, streams_.size());
+    if (rle) {
+      // Write RLE encoding marker.
+      writeInt32(&out, kRLE.size());
+      out.write(kRLE.data(), kRLE.size());
+      // Write number of RLE values.
+      writeInt32(&out, numRows);
+    }
+
+    for (auto& stream : streams_) {
+      stream->flush(&out);
+    }
+    const int32_t uncompressedSize = out.tellp();
+    VELOX_CHECK_LE(
+        uncompressedSize,
+        codec_->maxUncompressedLength(),
+        "UncompressedSize exceeds limit");
+    auto compressed = codec_->compress(out.getIOBuf().get());
+    const int32_t compressedSize = compressed->length();
+    writeInt32(output, uncompressedSize);
+    writeInt32(output, compressedSize);
+    const int32_t crcOffset = output->tellp();
+    writeInt64(output, 0); //  // Write zero checksum
+    // Number of columns and stream content. Unpause CRC.
+    if (listener) {
+      listener->resume();
+    }
+    output->write(
+        reinterpret_cast<const char*>(compressed->writableData()),
+        compressed->length());
+    // Pause CRC computation
+    if (listener) {
+      listener->pause();
+    }
+    const int32_t endSize = output->tellp();
+    // Fill in crc
+    int64_t crc = 0;
+    if (listener) {
+      crc = computeChecksum(listener, codec, numRows, compressedSize);
+    }
+    output->seekp(crcOffset);
+    writeInt64(output, crc);
+    output->seekp(endSize);
+  }
+
+  // Writes the contents to 'stream' in wire format
+  void flushInternal(int32_t numRows, bool rle, OutputStream* out) {
+    auto listener = dynamic_cast<PrestoOutputStreamListener*>(out->listener());
+    // Reset CRC computation
+    if (listener) {
+      listener->reset();
+    }
+
+    if (!needCompression(*codec_)) {
+      flushUncompressed(numRows, rle, out, listener);
+    } else {
+      flushCompressed(numRows, rle, out, listener);
```

Comment:
If flushUncompressed and flushCompressed are only used internally shall we make them private?